### PR TITLE
Old Gods Sacrifice Decisions Rework

### LINKED ISSUE #848: Secret OG cultists can sacrifice characters without raising any suspicion
**Mod Version**
86a5913f

**What expansions do you have installed?**
All

**Please explain your issue in as much detail as possible:**
Secret cultists can sacrifice characters to offmap old gods without any suspicion 

**Steps to reproduce the issue:**
Sacrifice many character to any OG

**Upload an attachment below: .zip of your save, or screenshots:**

<details>
<summary>Click to expand</summary>

image

</details>

### PR DESCRIPTION
**Pull request by MemeWizard**
#### Changelog:
- `Sacrifice to Void Lords` society decision and `Sacrifice to Old God` offmap decision were merged into one (with special decision icon, yay!). The new decision appears in two places - OG offmap screen and character interactions with prisoners.
- The character can use the decision if it:
  - Truly worships Old Gods (either their secret religion is OG or they publicly worship OG and don't have secret religion).
  - One of the following: 
    - Publicly worships OG.
    - Is a member of the Twilight Hammer cult.
- If the character publicly worships OG, it gets piety from sacrifices. If the character is a member of the Twilight Hammer cult, however, it doesn't get piety even if it publicly worships OG.
- Both Hakkari and OG now can sacrifice children and dragons
- Fixes #848

### CODE REVIEW

File: localisation/00_wc_vanilla_replacements.csv [hunk 1]
```diff
@@ -227,10 +227,10 @@ HF_211_DESC;Lying in the crib, I hear a slithering presence drawing closer and c
...
+EVTDESCHF23001;[target_localization_sacrifice.GetTitledFirstName] was sacrificed to [FromFrom.Religion.GetHighGodName]! Accounts describe how [target_localization_sacrifice.GetSheHe] was forcefully dragged screaming on top of a temple and torn apart by mad priests.;;;;;;;;;;;;;x
+EVTDESC_SSI_060_B;[From.GetTitledName] was sacrified to [FromFrom.Religion.GetHighGodName] by [FromFrom.GetFirstName]'s priests! After bringing [From.GetHerHim] to one of their [FromFrom.GetHouseOfWorship], they placed [From.GetFirstName] on an altar and sliced [From.GetHerHim] open with a ceremonial dagger. They then tore out [From.GetHerHis] still beating heart to the cheers of the assembled crowd.;;;;;;;;;;;;;x
+EVTDESC_SSI_072;Your priests have sacrificed [From.GetTitledName] to [Root.Religion.GetHighGodName]! After bringing [From.GetHerHim] to our [Root.GetHouseOfWorship], the priests placed [From.GetFirstName] on an altar and sliced [From.GetHerHim] open with a ceremonial dagger. They then tore out [From.GetHerHis] still beating heart to the cheers of the assembled crowd.;;;;;;;;;;;;;x
+EVTOPTA_SSI_072;[Root.Religion.GetHighGodNameCap] will be pleased.;;;;;;;;;;;;;x
```
Comment (arithon) [inline localisation/00_wc_vanilla_replacements.csv]: Is this really necessary? Aren't all God's names with a Capital? However, won't hurt to write it like you do, I suppose.

File: localisation/00_wc_vanilla_replacements.csv [hunk 2]
```diff
@@ -255,4 +255,16 @@ bloodthirsty_mass_sacrifice_bonus_3_desc;By appeasing [Root.Religion.GetHighGodN
 EVTDESCAHF23463;[From.GetTitledFirstName] has performed a great sacrifice, slaughtering hundreds of our own people to garner the favor of [From.Religion.GetHighGodName] during this war. With such a large offering, we are sure to prevail against our foes.;;;;;;;;;;;;;x
 EVTDESCAHF23464;[From.GetTitledFirstName] has performed a great sacrifice, slaughtering hundreds of our own people to garner the favor of [From.Religion.GetHighGodName] during this war. [From.GetHerHisCap] fanatic obsession with sacrifices will doom us all.;;;;;;;;;;;;;x
 EVTDESCAHF23465;[From.GetTitledFirstName] has performed a great sacrifice, slaughtering hundreds of [Root.GetHerHis] own people to garner the favor of [From.Religion.GetHighGodName] during this war. [From.GetSheHeCap] hopes that the Blood God will now grant [From.GetHerHis] armies the strength necessary to defeat my warriors.;;;;;;;;;;;;;x
-EVTOPTAHF23465;What if [From.Religion.GetHighGodName] listened to [From.GetHerHim]?;;;;;;;;;;;;;x
\ No newline at end of file
+EVTOPTAHF23465;What if [From.Religion.GetHighGodName] listened to [From.GetHerHim]?;;;;;;;;;;;;;x
```
Comment (arithon) [inline localisation/00_wc_vanilla_replacements.csv]: Did you change anything in these lines? Look pretty vanilly to me.

File: localisation/00_wc_vanilla_replacements.csv (hunk 1 above)
Comment (MemeWizard) [reply localisation/00_wc_vanilla_replacements.csv]: "the Old Gods will be pleased."

File: localisation/00_wc_vanilla_replacements.csv (hunk 2 above)
Comment (MemeWizard) [reply localisation/00_wc_vanilla_replacements.csv]: > What if the gods listened to [From.GetHerHim]?

Comment (arithon) [reply localisation/00_wc_vanilla_replacements.csv]: ok :)

File: localisation/00_wc_vanilla_replacements.csv [hunk 3]
```diff
@@ -227,10 +227,10 @@ HF_211_DESC;Lying in the crib, I hear a slithering presence drawing closer and c
...
-EVTDESC_SSI_072;Your priests have sacrificed [From.GetTitledName] to [Root.Religion.GetHighGodName]! After bringing [From.GetHerHim] to one of our temples, the priests placed [From.GetFirstName] on an altar and sliced [From.GetHerHim] open with a ceremonial dagger. They then tore out [From.GetHerHis] still beating heart to the cheers of the assembled crowd.;;;;;;;;;;;;;x
-EVTOPTA_SSI_072;[Root.Religion.GetHighGodName] will be pleased.;;;;;;;;;;;;;x
+EVTDESCHF23001;[target_localization_sacrifice.GetTitledFirstName] was sacrificed to [FromFrom.Religion.GetHighGodName]! Accounts describe how [target_localization_sacrifice.GetSheHe] was forcefully dragged screaming on top of a temple and torn apart by mad priests.;;;;;;;;;;;;;x
+EVTDESC_SSI_060_B;[From.GetTitledName] was sacrified to [FromFrom.Religion.GetHighGodName] by [FromFrom.GetFirstName]'s priests! After bringing [From.GetHerHim] to one of their [FromFrom.GetHouseOfWorship], they placed [From.GetFirstName] on an altar and sliced [From.GetHerHim] open with a ceremonial dagger. They then tore out [From.GetHerHis] still beating heart to the cheers of the assembled crowd.;;;;;;;;;;;;;x
```
Comment (zumbak04) [inline localisation/00_wc_vanilla_replacements.csv]: I'm not an expert, but `GetHouseOfWorship` is singular and it's incorrect to say `one of their <Singular Noun>`.

File: localisation/00_wc_vanilla_replacements.csv [hunk 4]
```diff
@@ -255,4 +255,16 @@ bloodthirsty_mass_sacrifice_bonus_3_desc;By appeasing [Root.Religion.GetHighGodN
...
..., soit � cause de la chaleur des b�chers voisins, soit de peur. Ses v�tements sacr�s pendent d�j� en lambeaux, exposant sa poitrine. Alors que l'heure pr�vue s'approche, le chant de vos fr�res s'accentue crescendo jusqu'� ce que, enfin, le moment soit venu.;[target_localization_sacrifice.GetTitledFirstName] ist an den Altar gefesselt und [target_localization_sacrifice.GetSheHe] scheint durch den Knebel Gebete zu murmeln. [target_localization_sacrifice.GetHerHisCap] K�rper ist schwei�nass, sei es aufgrund der Hitze der umliegenden Scheiterhaufen oder aus Angst. Das heilige Gewand h�ngt [target_localization_sacrifice.Getihmihr] in Fetzen von der Brust. Als die angesetzte Stunde n�her r�ckt, w�chst der Gesang Eurer Br�der zu einem Crescendo heran, bis es endlich so weit ist.;;Han atado a [target_localization_sacrifice.GetTitledFirstName] al altar. Crees o�r sus s�plicas a trav�s de la mordaza. Est� sudando como un animal por el miedo o por las piras encendidas a su alrededor. Sus santas vestiduras han quedado reducidas a jirones y su pecho ha quedado al descubierto. Se acerca la hora prevista y los c�nticos de los miembros de la orden llegan a un punto �lgido. Es el momento.;;;;;;;;;x
```
Comment (zumbak04) [inline localisation/00_wc_vanilla_replacements.csv]: Just an idea. Could you copypaste vanilla lines on the master, so we can see what you changed?

File: decisions/wc_united_sacrifice_decision.txt [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+		}
+
+		effect = {
+			FROM = {
+				show_scope_change = no
+
+				ROOT = {
+					show_scope_change = no
+
+					caltulate_PREV_offmap_old_gods_currency_insanity_effect = yes
```
Comment (zumbak04) [inline decisions/wc_united_sacrifice_decision.txt]: Why it's used outside `PREV_sacrifice_THIS_to_old_god_effect`? It's always used together.

File: decisions/wc_offmap_decisions.txt [hunk 6]
```diff
@@ -46,19 +40,21 @@ offmap_decisions = {
...
 			}
 		}
-		
+
 		third_party_allow = {
 			#show_only_failed_conditions = yes
 		}
 		
-		third_party_score = {
+		third_party_score = { # DO NOT FORGET TO CHANGE "caltulate_PREV_offmap_old_gods_currency_insanity_effect" IF YOU CHANGE THIS!
```
Comment (zumbak04) [inline decisions/wc_offmap_decisions.txt]: Please write a comment that it's no longer used in calculations, just shows stuff.

**Comment by zumbak04** [comment]
You get more impiety when you sacrifice via the offmap interface because of scopes.
<details>
<summary>Offmap</summary>
https://cdn.discordapp.com/attachments/340859955957596171/678730335810486305/unknown.png
</details>
<details>
<summary>Not offmap</summary>
https://cdn.discordapp.com/attachments/340859955957596171/678731173433704488/unknown.png
</details>

**Comment by MemeWizard** [comment]
> You get more impiety when you sacrifice via the offmap interface because of scopes.

Ok, I think I need to rebase branches a bit, because I'm sure I fixed that on one of the other branches

File: localisation/00_wc_vanilla_replacements.csv (hunk 3 above)
Comment (arithon) [reply localisation/00_wc_vanilla_replacements.csv]: You're right. Should add "s" after the <>.

Comment (zumbak04) [reply localisation/00_wc_vanilla_replacements.csv]: Better get rid of the construction and just use singular form. You can't be sure plural form will look like this.

File: common/scripted_effects/wc_sacrifice_scripted_effects.txt [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+	}
+}
+
+PREV_sacrifice_THIS_to_old_god_effect = {
+	if = {
+		limit = { PREV = { NOT = { society_member_of = twilights_hammer_cult } } }
+		PREV_public_sacrifice_THIS_effect = yes
+	}
+	else_if = {
+		limit = { PREV = { society_member_of = twilights_hammer_cult } }
```
[the remark this answers is not in the document]
Comment (MemeWizard) [reply common/scripted_effects/wc_sacrifice_scripted_effects.txt]: Fixed

File: localisation/00_wc_vanilla_replacements.csv (hunk 3 above)
Comment (MemeWizard) [reply localisation/00_wc_vanilla_replacements.csv]: Fixed

File: decisions/wc_offmap_decisions.txt (hunk 6 above)
Comment (MemeWizard) [reply decisions/wc_offmap_decisions.txt]: Added

File: decisions/wc_united_sacrifice_decision.txt (hunk 5 above)
Comment (MemeWizard) [reply decisions/wc_united_sacrifice_decision.txt]: Good catch

**Comment by MemeWizard** [comment]
> You get more impiety when you sacrifice via the offmap interface because of scopes.

> Ok, I think I need to rebase branches a bit, because I'm sure I fixed that on one of the other branches

@zumbak04, yep problem was already fixed

<details>
<summary>Click to expand</summary>

image
image

</details>

File: localisation/00_wc_vanilla_replacements.csv (hunk 4 above)
Comment (MemeWizard) [reply localisation/00_wc_vanilla_replacements.csv]: Pasted vanilla lines in #1181, so, please, check it in tortoise

<details>
<summary>Click to expand</summary>

image

</details>

**Comment by zumbak04** [comment]
Sacrificing to Hakkar doesn't give piety.
<details>
<summary>Click to expand</summary>

image

</details>

**Comment by zumbak04** [comment]
> Sacrificing to Hakkar doesn't give piety.

Ma bad. I see the decision gives it now.

File: common/scripted_effects/wc_sacrifice_scripted_effects.txt [hunk 8]
```diff
@@ -121,59 +209,46 @@ caltulate_PREV_offmap_old_gods_currency_insanity_effect = {
 }
 
 add_offmap_old_gods_currency_insanity_after_calculation_effect = {
-	if = {
-		limit = { old_gods_offmap_decisions_enabled_trigger = yes }
-
+	PREV = {
```
Comment (zumbak04) [inline common/scripted_effects/wc_sacrifice_scripted_effects.txt]: Why `PREV` is used here? From what I see it's needed only inside `PREV_sacrifice_THIS_to_old_god_effect`. Other places where `add_offmap_old_gods_currency_insanity_after_calculation_effect` is used don't have `PREV` scope at all.

File: common/scripted_effects/wc_sacrifice_scripted_effects.txt [hunk 9]
```diff
@@ -1,4 +1,92 @@
...
+	any_liege = {
+		limit = {
+			ai = no
+			NOT = { character = PREVPREV }
+		}
+		character_event = { id = SSI.60 }
+	}
+	PREV = {
+		if = {
+			limit = { NOT = { has_character_flag = wc_do_not_show_post_sacrifice_ping_event_flag } }
```
Comment (zumbak04) [inline common/scripted_effects/wc_sacrifice_scripted_effects.txt]: Is it intended? The flag is never set.

Comment (MemeWizard) [reply common/scripted_effects/wc_sacrifice_scripted_effects.txt]: Yep, it;s used in #1126

File: common/scripted_effects/wc_sacrifice_scripted_effects.txt (hunk 8 above)
Comment (MemeWizard) [reply common/scripted_effects/wc_sacrifice_scripted_effects.txt]: Fixed

**Comment by zumbak04** [comment]
This event mentions the wrong person. I sacrifaced Gaiman, not Skeram.
<details>
<summary>Click to expand</summary>

image

</details>

File: decisions/wc_united_sacrifice_decision.txt [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+				}
+			}
+		}
+
+		ai_will_do = {
+			factor = 1
+			modifier = {
+				factor = 0
+				FROM = { trait = cynical }
+			}
```
Comment (zumbak04) [inline decisions/wc_united_sacrifice_decision.txt]: I would remove or change it. The Old Gods give you power, so you can sacrifice people without zeal for your own interests.

**Comment by MemeWizard** [comment]
> This event mentions the wrong person. I sacrifaced Gaiman, not Skeram.

Didn't noticed that MNM.7504 is actually in another file missing when copied lines, kek.

> Quest localization mentions Sacrifice to Void Lords though my decision is named differently.

Should be fixed now.

<details>
<summary>Click to expand</summary>

image
image

</details>

File: decisions/wc_united_sacrifice_decision.txt (hunk 10 above)
Comment (MemeWizard) [reply decisions/wc_united_sacrifice_decision.txt]: Can remove it, replace it with or add additional `content` trait requirement.
What will you pick?

Comment (zumbak04) [reply decisions/wc_united_sacrifice_decision.txt]: What if you're content, but zealous? :3

Comment (zumbak04) [reply decisions/wc_united_sacrifice_decision.txt]: Something like this?
```
modifier = {
	factor = 0
	FROM = {
		NOT = { trait = zealous }
		trait = content
	}
}
```

**Comment by zumbak04** [comment]
What if I'm a Hakkari secretly worshipping the Old Gods?! I don't see the sacrifice decision in the offmap interface. When I click on the characters, I can sacrifice to Hakkar only.
<details>
<summary>Click to expand</summary>

image
image

</details>

**Comment by zumbak04** [comment]
To Yogg-Saron? I follow N'zoth. Cho'gall follows Yogg-Saron though.
<details>
<summary>Click to expand</summary>

image
image
image


</details>

**Comment by MemeWizard** [comment]
> What if I'm a Hakkari secretly worshipping the Old Gods?! I don't see the sacrifice decision in the offmap interface. When I click on the characters, I can sacrifice to Hakkar only.

Changed conditions in order for player to understand better why their character can't sacrifice.

> To Yogg-Saron? I follow N'zoth. Cho'gall follows Yogg-Saron though. Wrong scopes?

`From` in decision and on society screen in power description is decision taker, but `From` on society screen in power name is society leader, kek.

**Comment by zumbak04** [comment]
Giant condition.
<details>
<summary>Click to expand</summary>

image

</details>

**Comment by MemeWizard** [comment]
@zumbak04, already fixed in #1209

**Comment by bantheadmin** [comment]
Old gods offmap sacrifice decision conditions bloated / don't fit to screen
<details>
<summary>Click to expand</summary>
Screen resolution: 1366x768

image
</details>

**[Unsure if intended]** As a member of Twilight's Hammer used offmap sacrifice decision to complete "sacrifice a ruler mission". Got "virgin sacrifice" text on what seems to be a generic prisoner sacrifice events (rewards 100 Void Gifts, 200 Insanity)

<details>
<summary>Click to expand</summary>
Screen resolution: 1366x768

image

</details>

**[Unsure if intended]** Another ruler sacrifice with no mission active (rewards 10 Void Gifts, 200 Insanity)

<details>
<summary>Click to expand</summary>
Screen resolution: 1366x768

image

</details>

When leaving cult and changing religion to non-OG, offmap power disappears.
Old gods offmap power **doesn't reappear** in any of these cases:
1. (Re)Adopt publicly OG religion
2. Rejoin cult
3. Continue as ruler's heir and join cult / adopt OG religion

Everything listed on the changelog works.

**Comment by MemeWizard** [comment]
> Old gods offmap sacrifice decision conditions bloated / don't fit to screen

Fixed in #1209

> When leaving cult and changing religion to non-OG, offmap power disappears.
Old gods offmap power doesn't reappear in any of these cases:

> 1. (Re)Adopt publicly OG religion
> 2. Rejoin cult
> 3. Continue as ruler's heir and join cult / adopt OG religion

Did your character pick one of the Old Gods as main god again after reconversion?

> Got "virgin sacrifice" text on what seems to be a generic prisoner sacrifice events

Same in vanilla

<details>
<summary>Click to expand</summary>

image
image

</details>

> Another ruler sacrifice with no mission active (rewards 10 Void Gifts

Yeah, intended

**Comment by sejtalk** [comment]
Targetted decision doesn't show the amount of insanity gained

**Comment by MemeWizard** [comment]
@sejtalk, It's Intended.
AFAIK, it's impossible to show the amount of offmap power without putting `third_party` in decision, which means it will require the amount of needed clicks and that kinda defeats purpose of this PR.
Also, event after sacrifice show how much Insanity your character have gained anyway.

**Comment by sejtalk** [comment]
wrong void lords
It should say old gods instead

**Comment by MemeWizard** [comment]
> It should say old gods instead

Since its considered being a demon-worshiping secret society, "Void Lords" is localization of twilight's hammer cult "satan". I didn't changed its name on this branch, so, the only thing I can suggest is create a new issue on github.

**Pull request merged**